### PR TITLE
strategy: ewo: add histogram

### PR DESCRIPTION
**Pull request by zenixls2**
- add ewo histogram described in the original tv script to help filtering signals.
- use atr * 3 + base line as TP/entry
image

### CODE REVIEW

**Comment by zenixls2** [comment]
@c9s ready to be merged.
- added emv but not used in trading
- add setting dump on exit
- update the comments
- refactor the source code

**Comment by zenixls2** [comment]
@c9s ready to get merged

File: pkg/strategy/ewoDgtrd/strategy.go [hunk 1]
```diff
@@ -1274,6 +1264,31 @@ func (s *Strategy) Run(ctx context.Context, orderExecutor bbgo.OrderExecutor, se
...
 
+		maString := "vwema"
+		if s.UseSma {
+			maString = "sma"
+		}
+		if s.UseEma {
+			maString = "ema"
+		}
+
+		color.HiYellow("----- EWO Settings -------")
```
Comment (c9s) [inline pkg/strategy/ewoDgtrd/strategy.go:1275]: ensure that color.X uses stderr, because optizmier use stdout to get the filename

**Review by c9s — APPROVED** [review]

File: config/ewo_dgtrd.yaml [hunk 2]
```diff
@@ -31,6 +31,12 @@ exchangeStrategies:
     ccistochFilterHigh: 80
     # CCI Stochastic Indicator low filter
     ccistochFilterLow: 20
+    # ewo change rate histogram's upperbound filter
```
Comment (c9s) [inline config/ewo_dgtrd.yaml:34]: `cciStochFilterLow` ? upper case the S?

File: config/ewo_dgtrd.yaml [hunk 3]
```diff
@@ -31,6 +31,12 @@ exchangeStrategies:
...
+    # set to 1 would intend to let all ewo pass
+    ewoChngFilterHigh: 1.
```
Comment (c9s) [inline config/ewo_dgtrd.yaml:36]: does Chng mean `change`? use `ewoChangeFilterHigh` instead?

File: pkg/strategy/ewoDgtrd/strategy.go (hunk 1 above)
Comment (zenixls2) [reply pkg/strategy/ewoDgtrd/strategy.go]: done

File: config/ewo_dgtrd.yaml (hunk 2 above)
Comment (zenixls2) [reply config/ewo_dgtrd.yaml:34]: done

File: config/ewo_dgtrd.yaml (hunk 3 above)
Comment (zenixls2) [reply config/ewo_dgtrd.yaml]: done

**Comment by zenixls2** [comment]
image
after rebase on the latest main

**Pull request merged**